### ISSUE TITLE
Incorrect policy loss

### ISSUE THREAD

**Issue by takuseno** [miyosuda/async_deep_reinforce#49]
Hi, I'm reading this repository to implement my own A3C. Then I found policy loss incorrect.

Current policy loss is at https://github.com/miyosuda/async_deep_reinforce/blob/master/game_ac_network.py#L31 .
```
# policy entropy
entropy = -tf.reduce_sum(self.pi * log_pi, reduction_indices=1)
      
 # policy loss (output)  (Adding minus, because the original paper's objective function is for gradient ascent, but we use gradient descent optimizer.)
policy_loss = - tf.reduce_sum( tf.reduce_sum( tf.multiply( log_pi, self.a ), reduction_indices=1 ) * self.td + entropy * entropy_beta )
```

So it's like `policy_loss = -log(pi) * a + beta * entropy`. In this case, `entropy` would be minimized.However, `entropy` should be maximized to avoid convergence for exploration. [Original paper](https://arxiv.org/pdf/1602.01783.pdf) says,

![image](https://user-images.githubusercontent.com/5235131/32336726-65c9ae4a-c033-11e7-9e0f-c755fbec9551.png)

Thus, correct policy loss should be `policy_loss = -log(pi) * a - beta * entropy`.

If I am wrong, just please close this issue. I hope this helps you improve this implementation.

@miyosuda

**Comment by takuseno** [comment]
Sorry, just I found I was wrong.
![image](https://user-images.githubusercontent.com/5235131/32337461-74102752-c035-11e7-8410-6f70818e788c.png)

Then negating the first line is following above equation.

I'm closing this issue.
